sources/api/openapi.yaml

```
@@ -324,3 +324,79 @@ paths:
           description: "Reboot requested"
         500:
           description: "Server error"
+
+  /actions/refresh-update:
+    post:
+      summary: "Query update repository and refresh list of updates"
+      operationId: "refresh_update"
+      responses:
+        204:
+          description: "Successful request"
+        500:
+          description: "Server error"
+        423:
+          description: "Update lock held. Try again in a moment"
+
+  /actions/prepare-update:
+    post:
+      summary: "Downloads the chosen update and write the update image to the inactive partition"
+      operationId: "prepare_update"
+      responses:
+        204:
+          description: "Successful request"
+        404:
+          description: "Chosen update does not exist"
+        409:
+          description: "Action not allowed according to current update state"
+        500:
+          description: "Server error"
+        423:
+          description: "Update lock held. Try again in a moment"
+
+  /actions/activate-update:
+    post:
+      summary: "Marks the inactive partition for boot to apply update"
```

```suggestion
      summary: "Mark the prepared update as active so you can reboot into the chosen version"
```
I'm not sure about this exact suggestion, but this one is important and I thought the wording might be slightly confusing.  I also think it's important to clarify that it's not *just* that the inactive partition is being activated, it's specifically that the inactive partition has the update that apiserver/t-b-u set up, because we're not allowing arbitrary bank-flip operations, just ones that follow this designated workflow; see comment below.